_targets.R

```diff
@@ -115,35 +117,128 @@ targets_list <- list(
-  tar_target(gcm_names, c('ACCESS', 'GFDL')),#, 'CNRM', 'IPSL', 'MRI', 'MIROC')),
-  tar_target(gcm_vars, c("evspsbl", "hfss")),
-  tar_target(gcm_dates, c('1999-01-01', '1999-01-15')),
+  tar_target(gcm_names, c('ACCESS', 'GFDL', 'CNRM', 'IPSL', 'MRI', 'MIROC')),
+  tar_target(gcm_dates_df,
+             tibble(
+               projection_period = c('1980_1999', '2040_2059', '2080_2099'),
+               start_datetime = c('1980-01-01 00:00:00', '2040-01-01 00:00:00', '2080-01-01 00:00:00'),
+               # Include midnight on the final day of the time period
+               end_datetime = c('2000-01-01 00:00:00', '2060-01-01 00:00:00', '2100-01-01 00:00:00')
+             )),
+  # Gathered manually from https://cida.usgs.gov/thredds/ncss/notaro_GFDL_2040_2059/dataset.html
+  tar_target(gcm_query_vars, c("pr", "tas", "qas", "rsns", "uas", "vas")), # missing longwave radiation
 
   # Download data from GDP for each tile & GCM name combination.
   # If the cells in a tile don't change, then the tile should not need to rebuild.
   tar_target(
     gcm_data_raw_feather,
     download_gcm_data(
-      out_file_template = "7_drivers_munge/tmp/7_GCM_%s_tile%s_raw.feather",
+      out_file_template = "7_drivers_munge/tmp/7_GCM_%s_%s_tile%s_raw.feather",
       query_geom = query_cells_centroids_list_by_tile,
       gcm_name = gcm_names,
-      query_vars = gcm_vars,
-      query_dates = gcm_dates
+      gcm_projection_period = gcm_dates_df$projection_period,
+      query_vars = gcm_query_vars,
+      query_dates = c(gcm_dates_df$start_datetime, gcm_dates_df$end_datetime)
     ),
     # TODO: might need to split across variables, too. Once we scale up, our queries
     # might be too large and chunking by variable could help.
-    pattern = cross(query_cells_centroids_list_by_tile, gcm_names),
-    format = "file"
+    pattern = cross(query_cells_centroids_list_by_tile, gcm_names, gcm_dates_df),
+    format = "file",
+    error = "continue"
   ),
 
   ##### Munge GDP output into NetCDF files that will feed into GLM #####
-  # TODO - munge data
+
+  # Need to munge GCM variables into useable GLM variables
+  tar_target(glm_vars_info,
+             tibble(
+               # TODO: MISSING LONGWAVE
+               var_name = c("Rain", "AirTemp", "Snow", "RelHum", "Shortwave", "Longwave", "WindSpeed"),
+               longname = c("Total daily rainfall",
+                            "Total daily snowfall derived from precipitation and air temp",
+                            "Average daily near surface air temperature",
+                            "Average daily percent relative humidity",
+                            "Average daily surface downward shortwave flux in air",
+                            "LONGWAVE EXPLANATION",
+                            "Average daily windspeed derived from anemometric zonal and anenometric meridional wind components"),
+               units = c("meters", "meters", "degrees Celcius", "", "???", "???", "???"),
```

Comment:
For GLM, we need these units:
| GLM driver input | [Units needed by GLM](https://gmd.copernicus.org/preprints/gmd-2017-257/gmd-2017-257-manuscript-version3.pdf) - see table at p59 |
| ----------- | ----------- | 
| Rain | m/day |
| Snow | m/day |
| AirTemp | degrees celsius |
| RelHum | percent ( 0 - 100) |
| Shortwave      | W/m2  |
| Longwave   | W/m2 |
| WindSpeed | m/s |

And see Jordan's notes about conversions [here](https://github.com/USGS-R/lake-temperature-process-models/issues/15#issuecomment-1006084022)